android/java/org/chromium/chrome/browser/brave_stats/BraveStatsUtil.java

```diff
@@ -163,25 +171,84 @@ public static void updateBraveShareStatsLayoutAndShare(View view) {
-
-        Intent shareIntent = Intent.createChooser(sendIntent, " ");
-        shareIntent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
-        context.startActivity(shareIntent);
+        Log.d("CRASHING", "shareStatsAction :");
+        try {
+            Context context = ContextUtils.getApplicationContext();
+            Bitmap bmp = convertToBitmap(view);
+            // ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+            // bmp.compress(Bitmap.CompressFormat.PNG, 100, byteArrayOutputStream);
+            Log.d("CRASHING", "getting path :");
+            
+            String path = "";
+
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+                path = MediaStore.Images.Media.insertImage(
+                    context.getContentResolver(), bmp, "tempimage", null);Log.d("CRASHING", "path :"+path);
+            } else {
+                storeImage(bmp);
+                path = getOutputMediaFile().getAbsolutePath();
+            }
+            Log.d("CRASHING", "path:"+getOutputMediaFile().getPath());
+            Log.d("CRASHING", "path:"+getOutputMediaFile().getAbsolutePath());
+            Uri uri = Uri.parse(path);Log.d("CRASHING", "uri :"+uri);
+
+            Intent sendIntent = new Intent();
+            sendIntent.setAction(Intent.ACTION_SEND);
+            sendIntent.putExtra(Intent.EXTRA_TEXT,
+                    context.getResources().getString(R.string.brave_stats_share_text));
+            sendIntent.putExtra(Intent.EXTRA_STREAM, uri);
+            sendIntent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+            sendIntent.setType("image/text");
+
+            Intent shareIntent = Intent.createChooser(sendIntent, " ");
+            shareIntent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+            context.startActivity(shareIntent);
+        } catch (Exception e){
+            Log.d("CRASHING", "exception :"+e);
+        }
+    }
+
+    private static void storeImage(Bitmap image) {
+        File pictureFile = getOutputMediaFile();
+        if (pictureFile == null) {
+            Log.d("CRASHING",
+                    "Error creating media file, check storage permissions: ");// e.getMessage());
+            return;
+        }
+        try {
+            FileOutputStream fos = new FileOutputStream(pictureFile);
+            image.compress(Bitmap.CompressFormat.PNG, 90, fos);
+            fos.close();
+        } catch (FileNotFoundException e) {
+            Log.d("CRASHING", "File not found: " + e.getMessage());
+        } catch (IOException e) {
+            Log.d("CRASHING", "Error accessing file: " + e.getMessage());
+        }
+    }
+
+        /** Create a File for saving an image or video */
+    private static File getOutputMediaFile(){
+        // To be safe, you should check that the SDCard is mounted
+        // using Environment.getExternalStorageState() before doing this.
+        File mediaStorageDir = new File(Environment.getExternalStorageDirectory()
```

Comment:
Are we checking for storage permission before this code block ? just making sure

Reply:
yes, before the share call 

https://github.com/brave/brave-core/blob/e963a3c3ad4b3803d0909f1aa44877a9d93adfe9/android/java/org/chromium/chrome/browser/shields/BraveShieldsHandler.java#L430